app/src/main/java/com/github/se/eventradar/model/data/EventDatabaseModule.kt

```diff
@@ -0,0 +1,20 @@
+package com.github.se.eventradar.model.data
+
+import com.github.se.eventradar.model.repository.event.FirebaseEventRepository
+import com.github.se.eventradar.model.repository.event.IEventRepository
+import com.github.se.eventradar.model.repository.event.MockEventRepository
+import dagger.Binds
+import dagger.Module
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+
+@Module
+@InstallIn(SingletonComponent::class)
+abstract class EventDatabaseModule {
+  @Binds
+  abstract fun bindFirebaseEventRepository(
+      userRepository: FirebaseEventRepository
+  ): IEventRepository
+
+  @Binds abstract fun bindMockEventRepository(userRepository: MockEventRepository): IEventRepository
+}
```

Comment:
Ok and here are a few things regarding the hilt modules (both event and user)
- first of all, I really like the method using `@Binds`. Some people differentiate between tests/app using `@Named` and it's quite ugly / impractical imo
- they are typically placed in a folder called `di` (for dependency injection)
- I am not the best with Hilt, but I don't think this will work (test it before, I might be wrong because I rarely worked with `@Binds`). To my understanding, you should have:
  - a module for the events in the `di` folder (e.g. `EventDatabaseModule`) with a `@Binds` and a `@Provides` (without the `@Provides`, Hilt will not know how to create an `IEventRepository` in this case)
  - a module for the events in the tests specifically (e.g. `EventDatabaseTestModule`) with a `@Binds` and a `@Provides`

> [!TIP]
> You can find an example [here](https://github.com/RaulinN/swent-snippets/blob/db561acbcbb3841b791343387e1441eb6229a69e/what-is-dependency-injection.kt#L73) (starting at line 73)

Reply:
Yes I think what I have won't work. I believe I can accomplish this just using `@Provides`.